### PR TITLE
[23.x] ci: replace Intel macOS CI job

### PR DESCRIPTION
**Pull request by fanquake**
Intel macOS jobs are no-longer available on Cirrus.

### CODE REVIEW

**Comment by fanquake** [comment]
https://github.com/bitcoin/bitcoin/pull/26879/checks?check_run_id=10602206141
```bash
Setting up g++-multilib (4:9.3.0-1ubuntu2) ...
Processing triggers for libc-bin (2.31-0ubuntu9.9) ...
bash: pip3: command not found

Exit status: 127
```

**Comment by MarcoFalke** [comment]
[ci: Remove unused package](https://github.com/bitcoin/bitcoin/pull/26879/commits/d5ba043a6ca29f92decb0cdae9c7c7a2350e55be) doesn't make sense on 23.x  (it is wrong)

**Comment by MarcoFalke** [comment]
Maybe the other https://github.com/bitcoin/bitcoin/pull/26234 commit needs to be dropped as well? (It shouldn't affect macos)

**Comment by MarcoFalke** [comment]
Alternatively the task can be removed or ignored. (We only backport changes that pass on the master CI, so something that breaks here but not on master for one task seems unlikely)

**Comment by hebasto** [comment]
Backporting of 8e017f32889e9536a436f40c91f34f24bfd10525 should fix the native macOS task.

**Comment by fanquake** [comment]
> Backporting of https://github.com/bitcoin/bitcoin/commit/8e017f32889e9536a436f40c91f34f24bfd10525

Added.

> Alternatively the task can be removed or ignored.

Yea. If this change doesn't work, happy to just remove it. The chance of some macOS only regression in this branch is basically nil, and I'd rather remove a consistently failing just than just leave it hanging around.

**Review by hebasto — APPROVED** [review]
~ACK c2ee70a89b332f217bfe330dd6bead916afe0814~

File: src/secp256k1/build-aux/ar-lib [hunk 1]
```diff
@@ -0,0 +1,271 @@
+#! /bin/sh
+# Wrapper for Microsoft lib.exe
```
Comment (MarcoFalke) [inline src/secp256k1/build-aux/ar-lib:2]: ????

Comment (fanquake) [reply src/secp256k1/build-aux/ar-lib]: Fixed overzealous use of `git add .`.

**Review by hebasto — APPROVED** [review]
ACK 821b454abb97d74a085af70f56c39da49ed06a5f

**Comment by MarcoFalke** [comment]
The metadata is wrong in the last commit. Also, could backport 0bb7a1f71db1d986ab824d114534fb7671024990 for easier cherrry-picking.

**Comment by MarcoFalke** [comment]
lgtm. (Did the cherry-pick locally and checked that there was no overall diff in the last push)

**Pull request merged**